server/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -101,9 +104,14 @@ public ChecksumBlobStoreFormat(String codec, String blobNameFormat, CheckedFunct
      * @param name          name to be translated into
      * @return parsed blob object
      */
-    public T read(BlobContainer blobContainer, String name, NamedXContentRegistry namedXContentRegistry) throws IOException {
+    public T read(BlobContainer blobContainer, String name, NamedXContentRegistry namedXContentRegistry,
+                  BigArrays bigArrays) throws IOException {
         String blobName = blobName(name);
-        return deserialize(blobName, namedXContentRegistry, Streams.readFully(blobContainer.readBlob(blobName)));
+        try (ReleasableBytesStreamOutput out = new ReleasableBytesStreamOutput(bigArrays);
+             InputStream in = blobContainer.readBlob(blobName)) {
+            Streams.copy(in, out, false);
```

Comment:
Gotta do the non-closing `copy` here and close the `in` via `try` because we can only close `out` once (multiple closes aren't idempotent and trip assertions.

Reply:
+1, `bytes()` should not be used once `ReleasableBytesStreamOutput` is closed